kmod/src/data.c

```diff
@@ -1915,22 +1915,23 @@ static int scoutfs_data_page_mkwrite(struct vm_area_struct *vma,
 	int err;
 
 	sb_start_pagefault(sb);
+	inode_lock(inode);
```

Comment:
I'd be surprised if this doesn't deadlock somewhere.  It certainly would in our code, but the bigger question is the rest of the kernel attempting to copy to userspace while holding the inode lock.  Either this needs to show why this is safe or, more likely, relax the inode lock requirement in get_block.  If it's just for serializing access to extents the extent_sem should do that.